### PR TITLE
chore: Deprecation for SmartEditModule in 4.0

### LINKED ISSUE #12168: Deprecation - Remove SmartEdit deprecated code replaced by feature lib

### PR DESCRIPTION
**Pull request by WeizhengSap**
close #12168

### CODE REVIEW

File: projects/schematics/src/migrations/4_0/constructor-deprecations/data/dynamic-attribute.service.migration.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const DYNAMIC_ATTRIBUTE_SERVICE_MIGRATION: ConstructorDeprecation = {
+  // projects/core/src/cms/services/dynamic-attribute.service.ts
+  class: DYNAMIC_ATTRIBUTE_SERVICE,
+  importPath: SPARTACUS_CORE,
+  deprecatedParams: [
+    {
+      className: SMART_EDIT_SERVICE,
```
Comment (bgambocjaviniar) [inline projects/schematics/src/migrations/4_0/constructor-deprecations/data/dynamic-attribute.service.migration.ts:15]: 👍 

was confused at first, but then I found it was only smart_edit_service that was always available before from https://github.com/SAP/spartacus/pull/10700/files#diff-52a4cc6780191074db65924bf0caa126312c9424153a983f73c7c703b3919753L10

File: projects/schematics/src/migrations/4_0/methods-and-properties-deprecations/methods-and-properties-deprecations.ts [hunk 2]
```diff
@@ -1,8 +1,13 @@
 import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
 import { MethodPropertyDeprecation } from '../../../shared/utils/file-utils';
 import { migrateMethodPropertiesDeprecation } from '../../mechanism/methods-and-properties-deprecations/methods-and-properties-deprecations';
+import { DYNAMIC_ATTRIBUTE_SERVICE_MIGRATION } from './data/dynamic-attribute.service.migration';
+import { PAGE_EVENT_BUILDER_MIGRATION } from './data/page-event.builder.ts.migration';
 
-export const METHOD_PROPERTY_DATA: MethodPropertyDeprecation[] = [];
+export const METHOD_PROPERTY_DATA: MethodPropertyDeprecation[] = [
+  ...PAGE_EVENT_BUILDER_MIGRATION,
```
Comment (bgambocjaviniar) [inline projects/schematics/src/migrations/4_0/methods-and-properties-deprecations/methods-and-properties-deprecations.ts:8]: 👍

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 3]
```diff
@@ -34,6 +35,19 @@ export const REMOVED_PUBLIC_API_DATA: DeprecatedNode[] = [
   //   importPath: SPARTACUS_STOREFRONTLIB,
   //   comment: `'${VARIANT_STYLE_SELECTOR_COMPONENT}' was moved to @spartacus/product/variants.`,
   // },
+
+  // projects/core/src/smart-edit/smart-edit.module.ts
+  {
+    node: 'SmartEditModule',
```
Comment (bgambocjaviniar) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: make these into constants as done above in other files `projects/schematics/src/shared/constants.ts`

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 4]
```diff
@@ -19,52 +27,143 @@ import {
...
-  //   importPath: SPARTACUS_STOREFRONTLIB,
-  //   comment: `'${VARIANT_STYLE_SELECTOR_COMPONENT}' was moved to @spartacus/product/variants.`,
-  // },
-
   // projects/core/src/smart-edit/smart-edit.module.ts
   {
-    node: 'SmartEditModule',
+    node: SMART_EDIT_MODULE,
     importPath: SPARTACUS_CORE,
     comment: `'SmartEditModule was removed. Use @spartacus/smartedit instead.`,
```
Comment (bgambocjaviniar) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: replace `'SmartEditModule` with `${SMART_EDIT_MODULE}`

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 5]
```diff
@@ -19,52 +27,143 @@ import {
...
   },
   // projects/core/src/smart-edit/services/smart-edit.service.ts
   {
-    node: 'SmartEditService',
+    node: SMART_EDIT_SERVICE,
     importPath: SPARTACUS_CORE,
     comment: `'SmartEditService was moved to @spartacus/smartedit/core.`,
```
Comment (bgambocjaviniar) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: replace `'SmartEditService` with `${SMART_EDIT_SERVICE}`

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]

**Review by bgambocjaviniar — APPROVED** [review]

**Comment by marlass** [comment]
Merged to `epic/4-0-deprecations`

**Pull request closed without merging**